includes/class-sensei.php

```diff
@@ -918,80 +919,203 @@ public function load_class( $class_name = '' ) {
 	} // End load_class()
 
 	/**
-	 * Filtering wp_count_comments to ensure that Sensei comments are ignored
+	 * Filtering wp_count_comments to ensure that Sensei comments are ignored.
 	 *
 	 * @since   1.4.0
 	 * @access  public
-	 * @param  array   $comments
-	 * @param  integer $post_id
-	 * @return array
-	 */
-	public function sensei_count_comments( $comments, $post_id ) {
-		global $wpdb;
+	 *
+	 * @param stdClass $comment_counts Counts by comment status.
+	 * @param int      $post_id        Post ID.
+	 *
+	 * @return stdClass
+	 */
+	public function sensei_count_comments( $comment_counts, $post_id ) {
+		if (
+			// If we are getting counts for a specific, non-Sensei post, return early.
+			(
+				! empty( $post_id )
+				&& ! in_array( get_post_type( $post_id ), [ 'course', 'lesson', 'quiz' ], true )
+			)
 
-		$post_id = (int) $post_id;
+			// If Sensei's comment counts aren't included, we don't need to adjust.
+			|| ! $this->comment_counts_include_sensei_comments( $post_id )
+		) {
+			return $comment_counts;
+		}
+
+		// If there are no Sensei comments to deduct, return early.
+		$sensei_counts = $this->get_sensei_comment_counts( $post_id );
+		if ( empty( $sensei_counts ) ) {
+			return $comment_counts;
+		}
 
-		$count = wp_cache_get( "comments-{$post_id}", 'counts' );
+		// If nothing else has provided a filter value, get the raw value and side-step this filter.
+		if ( empty( $comment_counts ) ) {
```

Comment:
Just as a note: right now, this is only executed if we're filtering `true` on `sensei_comment_counts_include_sensei_comments`. With WooCommerce, we will always have a filtered value at this point.

Reply:
I don't get this. So we have added the `sensei_comment_counts_include_sensei_comments` filter to allow others to notify us that there is an added filter which returns sensei comments (since the default one does not).

However the input `$comment_counts` is empty. Why do we need to populate it? Isn't this code purpose to just remove sensei comments?